modules/mobile/app/controllers/mobile/v0/payment_information_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'evss/ppiu/service'
+
+module Mobile
+  module V0
+    class PaymentInformationController < ApplicationController
+      before_action { authorize :evss, :access? }
+      before_action { authorize :ppiu, :access? }
+      before_action :validate_pay_info, only: :update
+      before_action(only: :update) { authorize(:ppiu, :access_update?) }
+
+      def index
+        payment_information = service.get_payment_information.responses[0]
+        render json: Mobile::V0::PaymentInformationSerializer.new(@current_user.id,
+                                                                  payment_information.payment_account,
+                                                                  payment_information.control_information)
+      end
+
+      def update
+        payment_information = service.update_payment_information(pay_info).responses[0]
```

Comment:
Should we check for success here? Or are failures handled by the controller stack?

Reply:
As far as I can tell all execution errors are caught and are ultimately returned with an appropriate HTTP status code. Additionally, based on the service we are using, I don't there would be any case where we would send the confirmation email without successfully updating.